src/docs/transactions.dox

```diff
@@ -220,6 +220,36 @@ transactions - the table below summarizes those constraints:
 | Commit  | Yes | durable_timestamp > stable_timestamp | Y | None |
 | Commit  | Yes | durable_timestamp != 0 \|\| commit_timestamp > stable_timestamp | N | If no durable timestamp is given when committing a prepared transaction, the commit timestamp must be greater than the stable timestamp. |
 
+@subsection timestamp_roundup Automatic rounding of timestamps
+Applications setting timestamps for a transaction have to comply with the
+constraints based on the global timestamp state. In order to be compliant with
+the constraints applications need to query the global timestamp state and
+check their timestamps for compliance and adjust timestamps if required.
+To simplify the burden on applications related to rounding up timestamps
+WiredTiger supports automatic rounding of timestamps in some scenarios.
+
+Applications can configure
+<code>roundup_timestamps=(prepared=true,read=true)</code> with
+WT_SESSION::begin_transaction.
+
+The configuration <code>roundup_timestamps=(prepared=true)</code> will be valid
+only for prepared transactions. It indicates that the prepare timestamp could be
+rounded up to the oldest timestamp, if the prepare timestamp is less than the
+oldest timestamp. This setting also indicates that the commit timestamp of the
+transaction could be rounded up to the prepare timestamp, if the commit
+timestamp is less than the prepare timestamp. Based on the timestamps values
+and constraints, enabling this configuration could result in only one of
+timestamps being rounded up. For example, for the timestamp values
+<code>prepare_timestamp=100, commit_timestamp=300, oldest_timestamp=200</code>
+with configuration <code>roundup_timestamps=(prepared=true)</code> only the
+prepare timestamp will be rounded up to the oldest timestamp and the commit
+timestamp will not be adjusted and the result will be
+<code>prepare_timestamp=200, commit_timestamp=300, oldest_timestamp=200</code>.
```

Comment:
Thanks for this. The one thing still missing in the documentation is what order things happen in when both need to be rounded up. Please consider adding a 2nd example where both need to be rounded up or modifying this one example to be that "worst" case.